extensions/pkg/controller/status.go

```diff
@@ -143,7 +143,8 @@ func (s *statusUpdater) ErrorCustom(ctx context.Context, obj extensionsv1alpha1.
 	// instead of adding key-value pairs ourselves here
 	s.logger.Error(fmt.Errorf(errDescription), "Error", s.logKeysAndValues(obj)...) //nolint:logcheck
 
-	lastOp, lastErr := ReconcileError(lastOperationType, errDescription, 50, gardencorev1beta1helper.ExtractErrorCodes(gardencorev1beta1helper.DetermineError(err, err.Error()))...)
+	// TODO(shafeeqes): Drop DetermineError usage here, once all the extensions have their own error code detection.
```

Comment:
I think, we should already drop `DetermineError` usage here, as the shoot flow tasks that wait for extension resources will still detect error codes from the error message.
We should only keep it there for a while and then drop it there (as outlined in https://github.com/gardener/gardener/issues/5444).

Reply:
> as the shoot flow tasks that wait for extension resources will still detect error codes from the error message.

hmm.. where is this done exactly? 
I couldn't find a usage of DetermineError or DetermineErrorCodes corresponding to it.